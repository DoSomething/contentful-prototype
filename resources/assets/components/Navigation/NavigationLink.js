import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';

const NavigationLink = props => (
  <NavLink
    {...props}
    className={classnames('nav-link', props.className)}
    activeClassName="is-active"
  >
    {props.children}
  </NavLink>
);

NavigationLink.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

NavigationLink.defaultProps = {
  className: null,
};

export default NavigationLink;
