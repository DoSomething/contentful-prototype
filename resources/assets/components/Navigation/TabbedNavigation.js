/* global window */

import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import { MEDIA_MEDIUM_SIZE_MIN } from '../../constants/media-sizes';
import './tabbed-navigation.scss';

class TabbedNavigation extends React.Component {
  /**
   * Class contructor
   * @param  {Object} props
   */
  constructor(props) {
    super(props);

    this.node = null;
    this.isAnimatingFrame = false;

    this.updateState = this.updateState.bind(this);
    this.requestFrame = this.requestFrame.bind(this);
    this.onScroll = this.onScroll.bind(this);

    this.state = {
      isStuck: false,
    };
  }

  /**
   * React lifecycle method called after render() method runs.
   */
  componentDidMount() {
    this.updateState();

    window.addEventListener('scroll', this.onScroll, false);
  }

  /**
   * Method called when the page is scrolled.
   */
  onScroll() {
    if (window.innerWidth <= MEDIA_MEDIUM_SIZE_MIN) {
      return;
    }

    this.requestFrame();
  }

  /**
   * Request frame of animation.
   */
  requestFrame() {
    if (! this.isAnimatingFrame) {
      window.requestAnimationFrame(this.updateState);
    }

    this.isAnimatingFrame = true;
  }

  /**
   * Update the component's state based on its location on the page.
   */
  updateState() {
    this.isAnimatingFrame = false;

    const tabbedNavRect = this.node.getBoundingClientRect();

    this.setState({ isStuck: tabbedNavRect.top <= 0 });
  }

  /**
   * Render the component.
   */
  render() {
    return (
      <div ref={node => (this.node = node)} id="tabbed-navigation" className={classnames('tabbed-navigation', { 'is-stuck': this.state.isStuck })}>
        <div className="wrapper">
          { this.props.children }
        </div>
      </div>
    );
  }
}

TabbedNavigation.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TabbedNavigation;
