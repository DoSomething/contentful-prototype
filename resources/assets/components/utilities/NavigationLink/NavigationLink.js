import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';

const NavigationLink = props => (
  <NavLink
    {...props}
    className={classnames('nav-link text-lg', props.className)}
    activeClassName="is-active"
  >
    {props.children}
  </NavLink>
);

NavigationLink.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

NavigationLink.defaultProps = {
  className: null,
};

export default NavigationLink;

// margin: 10px theme('spacing.3');
