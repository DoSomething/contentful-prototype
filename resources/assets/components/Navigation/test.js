import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TabbedNavigation from './TabbedNavigation';
import NavigationLink from './NavigationLink';

test('TabbedNavigation snapshot test', () => {
  const component = shallow(
    <TabbedNavigation>
      <NavigationLink>Awesome Link</NavigationLink>
    </TabbedNavigation>
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
