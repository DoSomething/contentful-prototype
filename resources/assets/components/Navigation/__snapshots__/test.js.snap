// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedNavigation snapshot test 1`] = `
<div
  className="tabbed-navigation"
  id="tabbed-navigation"
>
  <div
    className="wrapper"
  >
    <NavigationLink
      className={null}
    >
      Awesome Link
    </NavigationLink>
  </div>
</div>
`;
